feat: Addition of 'Practice Problems' section to sidebar with topic-specific question pages

Chaitanya-45 opened the pull request:
Fixes: #46 
- Added a new 'Practice Problems' section in the sidebar
- Created individual pages for questions covering various C++ topics:
  - Introductory Problems
  - Control Structure Problems
  - Arrays
  - Strings
  - Functions
  - Structures and Pointers
  - Sorting
  - Standard Template Library (STL)

subhadipbhowmik commented:
> Fixes: #46
> 
>     * Added a new 'Practice Problems' section in the sidebar
> 
>     * Created individual pages for questions covering various C++ topics:
>       
>       * Introductory Problems
>       * Control Structure Problems
>       * Arrays
>       * Strings
>       * Functions
>       * Structures and Pointers
>       * Sorting
>       * Standard Template Library (STL)

Could You Please Implement in the Blog section not in Docs?

Chaitanya-45 commented:
@subhadipbhowmik Please check the update, I've moved the practice problems to the blogs section and added respective tags categorizing them based on the concept and coding platform.

subhadipbhowmik commented on blog/2024-06-03-STL-problems/STL-problems.md:
```diff
@@ -0,0 +1,27 @@
+---
+slug: STL-problems
```
in slug make STL to lowercase

subhadipbhowmik commented on blog/2024-06-03-structure-probelms/structure-problems.md:
```diff
@@ -0,0 +1,15 @@
+slug: structure-problems
+title: Structures and Pointers Problems
+authors: [chaitanya]
+tags: ["30daysofcpp", "struct", "pointer","practice problems","hackerrank","geeksforgeeks","codechef"]
```
no space in tags use -

Chaitanya-45 commented:
@subhadipbhowmik I've made the suggested changes, please check.

subhadipbhowmik commented:
> @subhadipbhowmik I've made the suggested changes, please check.



> @subhadipbhowmik I've made the suggested changes, please check.

Just wait a while
